[Unified Text Replacement] Move `UnifiedTextReplacementController` from WebKit to WebCore

rr-codes opened the pull request:
#### e4bbed49b0933c9ebcb74df0975d37394165bc8b
<pre>
[Unified Text Replacement] Move `UnifiedTextReplacementController` from WebKit to WebCore
<a href="https://bugs.webkit.org/show_bug.cgi?id=275289">https://bugs.webkit.org/show_bug.cgi?id=275289</a>
<a href="https://rdar.apple.com/129438122">rdar://129438122</a>

Reviewed by Aditya Keerthi.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/dom/DocumentMarker.h:
* Source/WebCore/page/ChromeClient.h:
(WebCore::ChromeClient::textReplacementSessionShowInformationForReplacementWithIDRelativeToRect):
(WebCore::ChromeClient::textReplacementSessionUpdateStateForReplacementWithID):
(WebCore::ChromeClient::removeTextIndicatorStyleForID):
(WebCore::ChromeClient::cleanUpTextStylesForSessionID):
(WebCore::ChromeClient::addSourceTextIndicatorStyle):
(WebCore::ChromeClient::addDestinationTextIndicatorStyle):
* Source/WebCore/page/Page.cpp:
(WebCore::m_unifiedTextReplacementController):
(WebCore::Page::willBeginTextReplacementSession):
(WebCore::Page::didBeginTextReplacementSession):
(WebCore::Page::textReplacementSessionDidReceiveReplacements):
(WebCore::Page::textReplacementSessionDidUpdateStateForReplacement):
(WebCore::Page::didEndTextReplacementSession):
(WebCore::Page::textReplacementSessionDidReceiveTextWithReplacementRange):
(WebCore::Page::updateStateForSelectedReplacementIfNeeded):
(WebCore::Page::textReplacementSessionDidReceiveEditAction):
(WebCore::m_historyItemClient): Deleted.
* Source/WebCore/page/Page.h:
(WebCore::Page::unifiedTextReplacementController const):
* Source/WebCore/page/unified-text-replacement/UnifiedTextReplacementController.h: Added.
* Source/WebCore/page/unified-text-replacement/UnifiedTextReplacementController.mm: Added.
(WebCore::UnifiedTextReplacementController::characterRange):
(WebCore::UnifiedTextReplacementController::characterCount):
(WebCore::UnifiedTextReplacementController::resolveCharacterRange):
(WebCore::UnifiedTextReplacementController::plainText):
(WebCore::UnifiedTextReplacementController::UnifiedTextReplacementController):
(WebCore::contextRangeForDocument):
(WebCore::UnifiedTextReplacementController::willBeginTextReplacementSession):
(WebCore::UnifiedTextReplacementController::didBeginTextReplacementSession):
(WebCore::UnifiedTextReplacementController::textReplacementSessionDidReceiveReplacements):
(WebCore::UnifiedTextReplacementController::textReplacementSessionDidUpdateStateForReplacement):
(WebCore::UnifiedTextReplacementController::didEndTextReplacementSession&lt;UnifiedTextReplacement::Session::ReplacementType::PlainText&gt;):
(WebCore::UnifiedTextReplacementController::didEndTextReplacementSession&lt;UnifiedTextReplacement::Session::ReplacementType::RichText&gt;):
(WebCore::UnifiedTextReplacementController::didEndTextReplacementSession):
(WebCore::UnifiedTextReplacementController::textReplacementSessionDidReceiveTextWithReplacementRange):
(WebCore::UnifiedTextReplacementController::textReplacementSessionDidReceiveEditAction):
(WebCore::UnifiedTextReplacementController::textReplacementSessionPerformEditActionForPlainText):
(WebCore::UnifiedTextReplacementController::textReplacementSessionPerformEditActionForRichText):
(WebCore::UnifiedTextReplacementController::updateStateForSelectedReplacementIfNeeded):
(WebCore::UnifiedTextReplacementController::contextRangeForSessionWithID const):
(WebCore::UnifiedTextReplacementController::document const):
(WebCore::UnifiedTextReplacementController::findReplacementMarkerByID const):
(WebCore::UnifiedTextReplacementController::findReplacementMarkerContainingRange const):
(WebCore::UnifiedTextReplacementController::replaceContentsOfRangeInSessionInternal):
(WebCore::UnifiedTextReplacementController::replaceContentsOfRangeInSession):
* Source/WebCore/page/unified-text-replacement/UnifiedTextReplacementTypes.h: Renamed from Source/WebKit/Shared/WebUnifiedTextReplacementSessionData.h.
* Source/WebCore/platform/Logging.h:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Platform/Logging.h:
* Source/WebKit/Scripts/webkit/messages.py:
(headers_for_type):
* Source/WebKit/Shared/TextIndicatorStyle.serialization.in: Renamed from Source/WebKit/Shared/UnifiedTextReplacement.serialization.in.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/Shared/WebTextReplacementData.h: Removed.
* Source/WebKit/Shared/WebUnifiedTextReplacementContextData.h: Removed.
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/API/APIPageConfiguration.h:
(API::PageConfiguration::unifiedTextReplacementBehavior const):
(API::PageConfiguration::setUnifiedTextReplacementBehavior):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfiguration.mm:
(convertToPlatform):
(convertToWeb):
(-[WKWebViewConfiguration _setUnifiedTextReplacementBehavior:]):
(-[WKWebViewConfiguration _unifiedTextReplacementBehavior]):
(convertToPlatformBehavior): Deleted.
(convertToWebBehavior): Deleted.
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
* Source/WebKit/UIProcess/API/mac/WKWebViewMac.mm:
* Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.h:
* Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.mm:
(WebKit::PageClientImplCocoa::textReplacementSessionShowInformationForReplacementWithIDRelativeToRect):
(WebKit::PageClientImplCocoa::textReplacementSessionUpdateStateForReplacementWithID):
(WebKit::PageClientImplCocoa::textReplacementSessionShowInformationForReplacementWithUUIDRelativeToRect): Deleted.
(WebKit::PageClientImplCocoa::textReplacementSessionUpdateStateForReplacementWithUUID): Deleted.
* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
(WebKit::WebPageProxy::willBeginTextReplacementSession):
(WebKit::WebPageProxy::didBeginTextReplacementSession):
(WebKit::WebPageProxy::textReplacementSessionDidReceiveReplacements):
(WebKit::WebPageProxy::textReplacementSessionDidUpdateStateForReplacement):
(WebKit::WebPageProxy::didEndTextReplacementSession):
(WebKit::WebPageProxy::textReplacementSessionDidReceiveTextWithReplacementRange):
(WebKit::WebPageProxy::textReplacementSessionDidReceiveEditAction):
(WebKit::WebPageProxy::textReplacementSessionShowInformationForReplacementWithIDRelativeToRect):
(WebKit::WebPageProxy::textReplacementSessionUpdateStateForReplacementWithID):
(WebKit::WebPageProxy::textReplacementSessionShowInformationForReplacementWithUUIDRelativeToRect): Deleted.
(WebKit::WebPageProxy::textReplacementSessionUpdateStateForReplacementWithUUID): Deleted.
* Source/WebKit/UIProcess/PageClient.h:
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView canPerformActionForWebView:withSender:]):
(-[WKContentView unifiedTextReplacementBehavior]):
* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::unifiedTextReplacementBehavior const):
(WebKit::WebViewImpl::canHandleSwapCharacters const):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::textReplacementSessionShowInformationForReplacementWithIDRelativeToRect):
(WebKit::WebChromeClient::textReplacementSessionUpdateStateForReplacementWithID):
(WebKit::WebChromeClient::removeTextIndicatorStyleForID):
(WebKit::WebChromeClient::cleanUpTextStylesForSessionID):
(WebKit::WebChromeClient::addSourceTextIndicatorStyle):
(WebKit::WebChromeClient::addDestinationTextIndicatorStyle):
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKit/WebProcess/WebPage/Cocoa/TextIndicatorStyleController.mm:
(WebKit::TextIndicatorStyleController::contextRangeForSessionWithID const):
(WebKit::TextIndicatorStyleController::contextRangeForTextIndicatorStyle const):
(WebKit::TextIndicatorStyleController::addSourceTextIndicatorStyle):
(WebKit::TextIndicatorStyleController::addDestinationTextIndicatorStyle):
(WebKit::TextIndicatorStyleController::createTextIndicatorForRange):
* Source/WebKit/WebProcess/WebPage/Cocoa/UnifiedTextReplacementController.mm: Removed.
* Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
(WebKit::WebPage::updateTextIndicatorStyleVisibilityForID):
(WebKit::WebPage::willBeginTextReplacementSession):
(WebKit::WebPage::didBeginTextReplacementSession):
(WebKit::WebPage::textReplacementSessionDidReceiveReplacements):
(WebKit::WebPage::textReplacementSessionDidUpdateStateForReplacement):
(WebKit::WebPage::didEndTextReplacementSession):
(WebKit::WebPage::textReplacementSessionDidReceiveTextWithReplacementRange):
(WebKit::WebPage::textReplacementSessionDidReceiveEditAction):
(WebKit::WebPage::textReplacementSessionShowInformationForReplacementWithIDRelativeToRect):
(WebKit::WebPage::textReplacementSessionUpdateStateForReplacementWithID):
(WebKit::WebPage::textReplacementSessionShowInformationForReplacementWithUUIDRelativeToRect): Deleted.
(WebKit::WebPage::textReplacementSessionUpdateStateForReplacementWithUUID): Deleted.
* Source/WebKit/WebProcess/WebPage/TextIndicatorStyleController.h:
* Source/WebKit/WebProcess/WebPage/UnifiedTextReplacementController.h: Removed.
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::didChangeSelection):
(WebKit::WebPage::cleanUpTextStylesForSessionID):
(WebKit::WebPage::addSourceTextIndicatorStyle):
(WebKit::WebPage::addDestinationTextIndicatorStyle):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
(WebKit::WebPage::textIndicatorStyleController): Deleted.
(WebKit::WebPage::unifiedTextReplacementController): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:

Canonical link: <a href="https://commits.webkit.org/279890@main">https://commits.webkit.org/279890@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d4152bbab32bb99676971096e705a3c026724a60

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/54845 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/34285 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/7424 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/58123 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/5576 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/57145 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/41830 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/5596 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/44420 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/60/builds/3778 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/56940 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/32400 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/47516 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/25547 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🧪 webkitpy~~](https://ews-build.webkit.org/#/builders/6/builds/54671 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/29190 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/4851 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/3717 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/51033 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/5071 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/59713 "Built successfully") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/30097 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/5219 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/51844 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/31234 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/47596 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/51256 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/32247 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8121 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/31022 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rr-codes commented on Source/WebCore/page/unified-text-replacement/UnifiedTextReplacementController.mm:
```diff
@@ -0,0 +1,777 @@
+    return selection.firstRange();
+}
+
+void UnifiedTextReplacementController::willBeginTextReplacementSession(const std::optional<UnifiedTextReplacement::Session>& session, CompletionHandler<void(const Vector<UnifiedTextReplacement::Context>&)>&& completionHandler)
```
debating if I should put this class inside the UTR namespace or not

pxlcoder commented on Source/WebCore/page/Page.cpp:
```diff
@@ -4829,4 +4832,46 @@ void Page::gamepadsRecentlyAccessed()
 #endif
 
+#if ENABLE(UNIFIED_TEXT_REPLACEMENT)
+void Page::willBeginTextReplacementSession(const std::optional<WebCore::UnifiedTextReplacement::Session>& session, CompletionHandler<void(const Vector<WebCore::UnifiedTextReplacement::Context>&)>&& completionHandler)
```
Lots of unnecessary `WebCore::` in the changes here and below.

pxlcoder commented on Source/WebCore/page/Page.h:
```diff
@@ -67,6 +67,10 @@
 #endif
 
+#if ENABLE(UNIFIED_TEXT_REPLACEMENT)
+#include "UnifiedTextReplacementController.h"
```
Can we forward declare?

pxlcoder commented on Source/WebCore/page/unified-text-replacement/UnifiedTextReplacementTypes.h:
```diff
@@ -25,33 +25,89 @@
 #pragma once
 
-#if ENABLE(UNIFIED_TEXT_REPLACEMENT)
+#if PLATFORM(COCOA)
```
Can you explain this change? `ENABLE` is generally preferred over platform conditionals.

rr-codes replied:
In my head this was to be more consistent with how TextExtractionTypes.h looks but I will put it back

pxlcoder commented on Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
```diff
@@ -6402,6 +6402,72 @@ header: <WebCore/TextExtractionTypes.h>
     Vector<WebCore::TextExtraction::Item> children;
 };
 
+#if PLATFORM(COCOA)
```
Ditto.

rr-codes committed:
[Unified Text Replacement] Move `UnifiedTextReplacementController` from WebKit to WebCore
https://bugs.webkit.org/show_bug.cgi?id=275289
rdar://129438122

Reviewed by Aditya Keerthi.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/dom/DocumentMarker.h:
* Source/WebCore/page/ChromeClient.h:
(WebCore::ChromeClient::textReplacementSessionShowInformationForReplacementWithIDRelativeToRect):
(WebCore::ChromeClient::textReplacementSessionUpdateStateForReplacementWithID):
(WebCore::ChromeClient::removeTextIndicatorStyleForID):
(WebCore::ChromeClient::cleanUpTextStylesForSessionID):
(WebCore::ChromeClient::addSourceTextIndicatorStyle):
(WebCore::ChromeClient::addDestinationTextIndicatorStyle):
* Source/WebCore/page/Page.cpp:
(WebCore::m_unifiedTextReplacementController):
(WebCore::Page::willBeginTextReplacementSession):
(WebCore::Page::didBeginTextReplacementSession):
(WebCore::Page::textReplacementSessionDidReceiveReplacements):
(WebCore::Page::textReplacementSessionDidUpdateStateForReplacement):
(WebCore::Page::didEndTextReplacementSession):
(WebCore::Page::textReplacementSessionDidReceiveTextWithReplacementRange):
(WebCore::Page::updateStateForSelectedReplacementIfNeeded):
(WebCore::Page::textReplacementSessionDidReceiveEditAction):
(WebCore::m_historyItemClient): Deleted.
* Source/WebCore/page/Page.h:
(WebCore::Page::unifiedTextReplacementController const):
* Source/WebCore/page/unified-text-replacement/UnifiedTextReplacementController.h: Added.
* Source/WebCore/page/unified-text-replacement/UnifiedTextReplacementController.mm: Added.
(WebCore::UnifiedTextReplacementController::characterRange):
(WebCore::UnifiedTextReplacementController::characterCount):
(WebCore::UnifiedTextReplacementController::resolveCharacterRange):
(WebCore::UnifiedTextReplacementController::plainText):
(WebCore::UnifiedTextReplacementController::UnifiedTextReplacementController):
(WebCore::contextRangeForDocument):
(WebCore::UnifiedTextReplacementController::willBeginTextReplacementSession):
(WebCore::UnifiedTextReplacementController::didBeginTextReplacementSession):
(WebCore::UnifiedTextReplacementController::textReplacementSessionDidReceiveReplacements):
(WebCore::UnifiedTextReplacementController::textReplacementSessionDidUpdateStateForReplacement):
(WebCore::UnifiedTextReplacementController::didEndTextReplacementSession<UnifiedTextReplacement::Session::ReplacementType::PlainText>):
(WebCore::UnifiedTextReplacementController::didEndTextReplacementSession<UnifiedTextReplacement::Session::ReplacementType::RichText>):
(WebCore::UnifiedTextReplacementController::didEndTextReplacementSession):
(WebCore::UnifiedTextReplacementController::textReplacementSessionDidReceiveTextWithReplacementRange):
(WebCore::UnifiedTextReplacementController::textReplacementSessionDidReceiveEditAction):
(WebCore::UnifiedTextReplacementController::textReplacementSessionPerformEditActionForPlainText):
(WebCore::UnifiedTextReplacementController::textReplacementSessionPerformEditActionForRichText):
(WebCore::UnifiedTextReplacementController::updateStateForSelectedReplacementIfNeeded):
(WebCore::UnifiedTextReplacementController::contextRangeForSessionWithID const):
(WebCore::UnifiedTextReplacementController::document const):
(WebCore::UnifiedTextReplacementController::findReplacementMarkerByID const):
(WebCore::UnifiedTextReplacementController::findReplacementMarkerContainingRange const):
(WebCore::UnifiedTextReplacementController::replaceContentsOfRangeInSessionInternal):
(WebCore::UnifiedTextReplacementController::replaceContentsOfRangeInSession):
* Source/WebCore/page/unified-text-replacement/UnifiedTextReplacementTypes.h: Renamed from Source/WebKit/Shared/WebUnifiedTextReplacementSessionData.h.
* Source/WebCore/platform/Logging.h:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Platform/Logging.h:
* Source/WebKit/Scripts/webkit/messages.py:
(headers_for_type):
* Source/WebKit/Shared/TextIndicatorStyle.serialization.in: Renamed from Source/WebKit/Shared/UnifiedTextReplacement.serialization.in.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/Shared/WebTextReplacementData.h: Removed.
* Source/WebKit/Shared/WebUnifiedTextReplacementContextData.h: Removed.
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/API/APIPageConfiguration.h:
(API::PageConfiguration::unifiedTextReplacementBehavior const):
(API::PageConfiguration::setUnifiedTextReplacementBehavior):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfiguration.mm:
(convertToPlatform):
(convertToWeb):
(-[WKWebViewConfiguration _setUnifiedTextReplacementBehavior:]):
(-[WKWebViewConfiguration _unifiedTextReplacementBehavior]):
(convertToPlatformBehavior): Deleted.
(convertToWebBehavior): Deleted.
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
* Source/WebKit/UIProcess/API/mac/WKWebViewMac.mm:
* Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.h:
* Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.mm:
(WebKit::PageClientImplCocoa::textReplacementSessionShowInformationForReplacementWithIDRelativeToRect):
(WebKit::PageClientImplCocoa::textReplacementSessionUpdateStateForReplacementWithID):
(WebKit::PageClientImplCocoa::textReplacementSessionShowInformationForReplacementWithUUIDRelativeToRect): Deleted.
(WebKit::PageClientImplCocoa::textReplacementSessionUpdateStateForReplacementWithUUID): Deleted.
* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
(WebKit::WebPageProxy::willBeginTextReplacementSession):
(WebKit::WebPageProxy::didBeginTextReplacementSession):
(WebKit::WebPageProxy::textReplacementSessionDidReceiveReplacements):
(WebKit::WebPageProxy::textReplacementSessionDidUpdateStateForReplacement):
(WebKit::WebPageProxy::didEndTextReplacementSession):
(WebKit::WebPageProxy::textReplacementSessionDidReceiveTextWithReplacementRange):
(WebKit::WebPageProxy::textReplacementSessionDidReceiveEditAction):
(WebKit::WebPageProxy::textReplacementSessionShowInformationForReplacementWithIDRelativeToRect):
(WebKit::WebPageProxy::textReplacementSessionUpdateStateForReplacementWithID):
(WebKit::WebPageProxy::textReplacementSessionShowInformationForReplacementWithUUIDRelativeToRect): Deleted.
(WebKit::WebPageProxy::textReplacementSessionUpdateStateForReplacementWithUUID): Deleted.
* Source/WebKit/UIProcess/PageClient.h:
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView canPerformActionForWebView:withSender:]):
(-[WKContentView unifiedTextReplacementBehavior]):
* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::unifiedTextReplacementBehavior const):
(WebKit::WebViewImpl::canHandleSwapCharacters const):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::textReplacementSessionShowInformationForReplacementWithIDRelativeToRect):
(WebKit::WebChromeClient::textReplacementSessionUpdateStateForReplacementWithID):
(WebKit::WebChromeClient::removeTextIndicatorStyleForID):
(WebKit::WebChromeClient::cleanUpTextStylesForSessionID):
(WebKit::WebChromeClient::addSourceTextIndicatorStyle):
(WebKit::WebChromeClient::addDestinationTextIndicatorStyle):
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKit/WebProcess/WebPage/Cocoa/TextIndicatorStyleController.mm:
(WebKit::TextIndicatorStyleController::contextRangeForSessionWithID const):
(WebKit::TextIndicatorStyleController::contextRangeForTextIndicatorStyle const):
(WebKit::TextIndicatorStyleController::addSourceTextIndicatorStyle):
(WebKit::TextIndicatorStyleController::addDestinationTextIndicatorStyle):
(WebKit::TextIndicatorStyleController::createTextIndicatorForRange):
* Source/WebKit/WebProcess/WebPage/Cocoa/UnifiedTextReplacementController.mm: Removed.
* Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
(WebKit::WebPage::updateTextIndicatorStyleVisibilityForID):
(WebKit::WebPage::willBeginTextReplacementSession):
(WebKit::WebPage::didBeginTextReplacementSession):
(WebKit::WebPage::textReplacementSessionDidReceiveReplacements):
(WebKit::WebPage::textReplacementSessionDidUpdateStateForReplacement):
(WebKit::WebPage::didEndTextReplacementSession):
(WebKit::WebPage::textReplacementSessionDidReceiveTextWithReplacementRange):
(WebKit::WebPage::textReplacementSessionDidReceiveEditAction):
(WebKit::WebPage::textReplacementSessionShowInformationForReplacementWithIDRelativeToRect):
(WebKit::WebPage::textReplacementSessionUpdateStateForReplacementWithID):
(WebKit::WebPage::textReplacementSessionShowInformationForReplacementWithUUIDRelativeToRect): Deleted.
(WebKit::WebPage::textReplacementSessionUpdateStateForReplacementWithUUID): Deleted.
* Source/WebKit/WebProcess/WebPage/TextIndicatorStyleController.h:
* Source/WebKit/WebProcess/WebPage/UnifiedTextReplacementController.h: Removed.
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::didChangeSelection):
(WebKit::WebPage::cleanUpTextStylesForSessionID):
(WebKit::WebPage::addSourceTextIndicatorStyle):
(WebKit::WebPage::addDestinationTextIndicatorStyle):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
(WebKit::WebPage::textIndicatorStyleController): Deleted.
(WebKit::WebPage::unifiedTextReplacementController): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:

Canonical link: https://commits.webkit.org/279890@main
